### ISSUE TITLE
Unable to run on Java 9 using sbt-protobuf

### ISSUE THREAD

**Issue by frohman04** [scalapb/ScalaPB#415]
```
addSbtPlugin("com.thesamet" % "sbt-protoc" % "0.99.16")

libraryDependencies += "com.thesamet.scalapb" %% "compilerplugin" % "0.7.1"
```

When I run the build on my build inside the sbt Docker container (hseeberger/scala-sbt:8u151-2.12.4-1.1.1, using Docker for Windows), everything works as expected:

```
[info] Compiling 1 protobuf files to /root/src/target/scala-2.12/src_managed/main
[info] Compiling schema /root/src/src/main/protobuf/wunderground.proto
protoc-jar: protoc version: 3.5.1, detected platform: linux-x86_64 (linux/amd64)
protoc-jar: embedded: bin/3.5.1/protoc-3.5.1-linux-x86_64.exe
protoc-jar: executing: [/tmp/protocjar4970673939057565723/bin/protoc.exe, --plugin=protoc-gen-scala=/tmp/protocbridge3192874169915983226, --scala_out=grpc:/root/src/target/scala-2.12/src_managed/main, -I/root/src/src/main/protobuf, -I/root/src/target/protobuf_external, -I/root/src/src/main/protobuf, -I/root/src/target/protobuf_external, /root/src/src/main/protobuf/wunderground.proto]
[info] Compiling protobuf
[info] Protoc target directory: /root/src/target/scala-2.12/src_managed/main
[info] Compiling 8 Scala sources to /root/src/target/scala-2.12/classes ...
[info] Non-compiled module 'compiler-bridge_2.12' for Scala 2.12.4. Compiling...
[info]   Compilation completed in 5.907s.
[info] Done compiling.
```

It also works natively on MacOS High Sierra with the latest Java 8.

However, when I run the build natively on my Windows 10 machine, I get the following exception:

```
[info] Compiling 1 protobuf files to C:\Users\frohman04\Documents\Projects\utility-stats\target\scala-2.12\src_managed\main
[info] Compiling schema C:\Users\frohman04\Documents\Projects\utility-stats\src\main\protobuf\wunderground.proto
protoc-jar: protoc version: 3.5.1, detected platform: windows-x86_64 (windows 10/amd64)
protoc-jar: embedded: bin/3.5.1/protoc-3.5.1-windows-x86_64.exe
protoc-jar: executing: [C:\Users\frohman04\AppData\Local\Temp\protocjar12032940230219557153\bin\protoc.exe, --plugin=protoc-gen-scala=C:\Users\frohman04\AppData\Local\Temp\protocbridge118128885610253061.bat, --scala_out=grpc:C:\Users\frohman04\Documents\Projects\utility-stats\target\scala-2.12\src_managed\main, -IC:\Users\frohman04\Documents\Projects\utility-stats\src\main\protobuf, -IC:\Users\frohman04\Documents\Projects\utility-stats\target\protobuf_external, -IC:\Users\frohman04\Documents\Projects\utility-stats\src\main\protobuf, -IC:\Users\frohman04\Documents\Projects\utility-stats\target\protobuf_external, C:\Users\frohman04\Documents\Projects\utility-stats\src\main\protobuf\wunderground.proto]
Error: Could not find or load main class protocbridge.frontend.BridgeApp
Caused by: java.lang.ClassNotFoundException: protocbridge.frontend.BridgeApp
--scala_out: protoc-gen-scala: Plugin failed with status code 1.
[error] java.lang.RuntimeException: protoc returned exit code: 1
[error]         at scala.sys.package$.error(package.scala:27)
[error]         at sbtprotoc.ProtocPlugin$.compile(ProtocPlugin.scala:142)
[error]         at sbtprotoc.ProtocPlugin$.compileProto$1(ProtocPlugin.scala:182)
[error]         at sbtprotoc.ProtocPlugin$.$anonfun$sourceGeneratorTask$4(ProtocPlugin.scala:186)
[error]         at sbt.util.FileFunction$.$anonfun$cached$1(FileFunction.scala:73)
[error]         at sbt.util.FileFunction$.$anonfun$cached$4(FileFunction.scala:147)
[error]         at sbt.util.Difference.apply(Tracked.scala:313)
[error]         at sbt.util.Difference.apply(Tracked.scala:293)
[error]         at sbt.util.FileFunction$.$anonfun$cached$3(FileFunction.scala:143)
[error]         at sbt.util.Difference.apply(Tracked.scala:313)
[error]         at sbt.util.Difference.apply(Tracked.scala:288)
[error]         at sbt.util.FileFunction$.$anonfun$cached$2(FileFunction.scala:142)
[error]         at sbtprotoc.ProtocPlugin$.$anonfun$sourceGeneratorTask$1(ProtocPlugin.scala:192)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:44)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:39)
[error]         at sbt.std.Transform$$anon$4.work(System.scala:66)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:262)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
[error]         at sbt.Execute.work(Execute.scala:271)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:262)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:174)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:36)
[error]         at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
[error]         at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
[error]         at java.base/java.lang.Thread.run(Unknown Source)
[error] (Compile / protocGenerate) protoc returned exit code: 1
```

I do have a somewhat interesting setup, where my user profile is stored on a second (D:) harddrive, and softlinked into the C:\Users directory.  I've tried to work around this by setting SBT_OPTS to have SBT reference the .ivy2 directory using the correct path on the D: drive to work around the symlink, as well as changing TEMP and TMP variables to also point directly to the D: drive.  I verified that both of these changes were reflected in the execution of the tool, but the build result was still the same.

**Comment by frohman04** [comment]
Finally figured out the issue.  My Windows system was using Java 9 instead of 8 like I thought.  After switching to using Java 8, the build worked again.  So, the problem seems to be caused by Java 9, not Windows.

**Comment by thesamet** [comment]
@augi will you be able to take a look? It seems like the problem is isolated to Java 9: https://ci.appveyor.com/project/thesamet/sbt-protoc/build/29  (I added the tests in https://github.com/thesamet/sbt-protoc/commit/9a5b4c78667ffd7f060c834179282b04da9cfeb4)

**Comment by thesamet** [comment]
@augi disregard the previous build failure, it is related to a Java 9 + SBT 0.13.x issue. I updated this project to sbt 1.1.1, and got the same failure on the original report with Java 9: https://ci.appveyor.com/project/thesamet/sbt-protoc/build/30/job/kuq1q1rtkuvjbsfm#L1067

**Comment by augi** [comment]
I hopefully look at this during this week. I've checked how `java` command works on Java 9 and [it should be OK](https://sites.google.com/a/athaydes.com/renato-athaydes/posts/guidetojava9-compilejarrun) to call it [as we are calling it now](https://github.com/scalapb/protoc-bridge/blob/master/src/main/scala/protocbridge/frontend/WindowsPluginFrontend.scala#L41).

Maybe [this PR](https://github.com/scalapb/protoc-bridge/pull/17) could be related (e.g. it could be problem with encoding).

**Comment by augi** [comment]
@thesamet I tried to execute the `test` of `protoc-bridge` in `jdk-9.0.4`, and they failed.

After applying [this fix](https://github.com/scalapb/protoc-bridge/pull/17), all the tests passed. Also tested in `jdk1.8.0_151`.

**Comment by thesamet** [comment]
I believe that this is resolved in sbt-protoc 0.99.18.
